akka-remote/src/main/scala/akka/remote/artery/SystemMessageDelivery.scala

```diff
@@ -176,6 +185,10 @@ private[akka] class SystemMessageDelivery(
           case msg @ (_: SystemMessage | _: AckedDeliveryMessage) ⇒
             if (unacknowledged.size < maxBufferSize) {
               seqNo += 1
+              if (unacknowledged.isEmpty)
+                ackTimestamp = System.nanoTime()
```

Comment:
shouldn't we reset this on every succesful delivery (when we remove elements from buffer) instead of when the buffer gets empty? It might be that messages are continuously delivered but the buffer does not get empty for a long time.